### PR TITLE
MP-CP 1.1

### PR DESCRIPTION
**Pull request by FroMage**

### CODE REVIEW

File: application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java [hunk 1]
```diff
@@ -8,6 +10,28 @@
 
 public class ApplicationContextProvider implements ThreadContextProvider {
 
+    static final ClassLoader SYSTEM_CL;
+
+    static {
+        final SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            SYSTEM_CL = AccessController
+                    .doPrivileged((PrivilegedAction<ClassLoader>) ApplicationContextProvider::calculateSystemClassLoader);
```
Comment (manovotn) [inline application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java:19]: @FroMage If we need privileged action to grab the CL, don't we also need it to set it (inside `currentContext` and `clearContext()`)?

Comment (FroMage) [reply application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java:19]: I just copied this from SR-Config, I didn't give it much thought.

Comment (manovotn) [reply application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java:19]: Got link to that SR config code?

Comment (FroMage) [reply application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java:19]: https://github.com/smallrye/smallrye-config/blob/master/implementation/src/main/java/io/smallrye/config/SmallRyeConfigProviderResolver.java#L47

Comment (manovotn) [reply application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java:19]: Ok, let's keep it this way for now. We can always fix it up later.

**Review by manovotn — APPROVED** [review]
Sorry for the delay, had some ISP issues and then I simply forgot about this PR...

**Comment by FroMage** [comment]
NP, thanks!

**Pull request merged**